<a href="https://colab.research.google.com/github/Noks06/Prediction-of-Stroke-and-Car-insurance-/blob/main/Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project 2 Part 1
by Nokuthula Mchunu

In [3]:
# mount the drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [5]:
#Importing relevent Liberaries
import pandas as pd
import numpy as np


##  First choice: Stroke Prediction Dataset

In [6]:
#read csv file and load to dataframe
strokefile ='/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week08/Data/healthcare-dataset-stroke-data.csv'

dfstroke = pd.read_csv(strokefile)
#explore top 5 rows
dfstroke.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [8]:
dfstroke.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


1. Source of data
  * The source of data is from Kaggle.com

2. Brief description of data
  * Dataset contain information of client or patiant to be predicted whether they have a stroke or not

3. What is the target?
  * Stroke
4. What does one row represent? (A person? A business? An event? A product?)
  * Row represent a person

5. Is this a classification or regression problem?
  * classification problem

6. How many features does the data have?
  * 12 columns
7. How many rows are in the dataset?
  * 5110 rows
8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
  * I dont foresee much challenges in cleaning the data, only have mbi column that have missing values that needs to be imputed, datatypes are in good shape, might have duplicates column and inconcestancy.Exploring the data there will be no challanges and with modelling may be the challenge will be to choose the best model out of many that I will be trying  

## Second choice: Car Insurance Dataset

In [11]:
# read csv file and load the dataframe
car_Insure_file ='/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week08/Data/Car_Insurance_Claim.csv'
dfcar_insure = pd.read_csv(car_Insure_file)

# explore top 5 record
dfcar_insure.head(5)

ID    AGE  GENDER      RACE DRIVING_EXPERIENCE    EDUCATION  \
0  569520    65+  female  majority               0-9y  high school   
1  750365  16-25    male  majority               0-9y         none   
2  199901  16-25  female  majority               0-9y  high school   
3  478866  16-25    male  majority               0-9y   university   
4  731664  26-39    male  majority             10-19y         none   

          INCOME  CREDIT_SCORE  VEHICLE_OWNERSHIP VEHICLE_YEAR  MARRIED  \
0    upper class      0.629027                1.0   after 2015      0.0   
1        poverty      0.357757                0.0  before 2015      0.0   
2  working class      0.493146                1.0  before 2015      0.0   
3  working class      0.206013                1.0  before 2015      0.0   
4  working class      0.388366                1.0  before 2015      0.0   

   CHILDREN  POSTAL_CODE  ANNUAL_MILEAGE VEHICLE_TYPE  SPEEDING_VIOLATIONS  \
0       1.0        10238         12000.0        sedan                    0   
1       0.0        10238         16000.0        sedan                    0   
2       0.0        10238         11000.0        sedan                    0   
3       1.0        32765         11000.0        sedan                    0   
4       0.0        32765         12000.0        sedan                    2   

   DUIS  PAST_ACCIDENTS  OUTCOME  
0     0               0      0.0  
1     0               0      1.0  
2     0               0      0.0  
3     0               0      0.0  
4     0               1      1.0

In [12]:
# get a quick stats on info
dfcar_insure.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 19 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   10000 non-null  int64  
 1   AGE                  10000 non-null  object 
 2   GENDER               10000 non-null  object 
 3   RACE                 10000 non-null  object 
 4   DRIVING_EXPERIENCE   10000 non-null  object 
 5   EDUCATION            10000 non-null  object 
 6   INCOME               10000 non-null  object 
 7   CREDIT_SCORE         9018 non-null   float64
 8   VEHICLE_OWNERSHIP    10000 non-null  float64
 9   VEHICLE_YEAR         10000 non-null  object 
 10  MARRIED              10000 non-null  float64
 11  CHILDREN             10000 non-null  float64
 12  POSTAL_CODE          10000 non-null  int64  
 13  ANNUAL_MILEAGE       9043 non-null   float64
 14  VEHICLE_TYPE         10000 non-null  object 
 15  SPEEDING_VIOLATIONS  10000 non-null  

1. Source of data
  * kaggle.com
2. Brief description of data
  * This data is for car insurance, predicting whether a client put a clain or not
3. What is the target?
  * Outcome
4. What does one row represent? (A person? A business? An event? A product?)
  * person
5. Is this a classification or regression problem?
  * classification problem
6. How many features does the data have?
  * Has 19 features
7. How many rows are in the dataset?
  * 10000 rows
8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

* I dont foresee much challenges in cleaning the data, only have credit_score and annual ANNUAL_MILEAGE column that have missing values that needs to be imputed, datatypes are in good shape, might have duplicates column and inconcestancy.Exploring the data there will be no challanges and with modelling may be the challenge will be to choose the best model out of many that I will be trying  